PythonTool/TagExpansion.py

```diff
@@ -168,20 +180,36 @@ def expand_vector_tags(mesh_file, main_dir_name = None):
        LookupError: If the file does not contain any vector tags.
     """
 
+    # Create a dictionary with the MB element types.
+    elements = {
+        "vertex" : types.MBVERTEX, "edge" : types.MBEDGE, "tri" : types.MBTRI,
+        "quad" : types.MBQUAD, "polygon" : types.MBPOLYGON, "tet" : types.MBTET,
+        "pyramid" : types.MBPYRAMID, "prism" : types.MBPRISM, "knife" : types.MBKNIFE,
+        "hex" : types.MBHEX, "polyhedron" : types.MBPOLYHEDRON, "maxtype" : types.MBMAXTYPE,
+        "entityset" : types.MBENTITYSET
+        }
+
     # Load the mesh file to create a reference mesh.
     mb_ref = core.Core()
     mb_ref.load_file(mesh_file)
 
-    # Retrieve the lists of scalar and vector tags on the mesh.
+    # Ensure the MB element type is valid.
+    mb_type = elements.get(element_type.lower())
+
+    if mb_type is None:
+        print("WARNING: Please choose a valid MB element type.")
+        exit()
+
+    # Retrieve the lists of scalar and vector tags on the reference mesh.
     try:
-        hexes_ref, scal_tags_ref, vec_tags_ref = get_tag_lists(mb_ref, types.MBHEX)
+        elements_ref, scal_tags_ref, vec_tags_ref = get_tag_lists(mb_ref, element_type, mb_type)
     except LookupError as e:
         print(str(e))
         exit()
```

Comment:
If you don't catch this error, won't it just print it anyway and exit?

Reply:
Yes, you're right! I fixed that, and also eliminated all exit() calls in my script.